### PR TITLE
pluggable graph execution

### PR DESCRIPTION
**Pull request by Shiti**
A h1st graph is a representation of execution flow similar to a flowchart. 
A h1st graph can be used at various stages of application lifecycle - data ingestion, data transformation, 
api definition, model training, etc. and requires execution capabilities. 

The graph definition and execution are separate so that the execution engine can be chosen without impacting user code.
graph

### CODE REVIEW

**Comment by TgithubJ** [comment]
@Shiti please add a simple example on 
1. how to build h1st modeler, h1st model
2. how to use them inside a graph

File: h1st/core/graph.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        is_root_node = to_node in self.__root_nodes
+        if is_root_node and len(self.__root_nodes) == 1:
+            raise RuntimeError("There must be one or more root nodes")
+        if from_node not in self.__graph.keys():
+            self.add_node(from_node)
+        if to_node not in self.__graph.keys():
+            self.add_node(to_node)
+        if from_node in self.__graph[to_node].next_nodes:
+            raise RuntimeError("Cannot add edge as it results in a closed loop")
+        if to_node not in self.__graph[from_node].next_nodes and \
```
Comment (phamhoangtuan) [inline h1st/core/graph.py:34]: this one can we use `elif to_node not in self.__graph[from_node].next_nodes` to reduce length of if condition?

**Comment by ctn** [comment]
What should we do with this PR?

**Comment by ctn** [comment]
Probably out-of-date.

**Pull request closed without merging**